### PR TITLE
Fixing CompositeMap object so that it respects masked maps

### PR DESCRIPTION
**Pull request by wafels**
Issue #2490 notes that masked maps are not rendered when using CompositeMap.  This PR fixes this by using the mask detection mechanism in Map and implementing it in CompositeMap.

### CODE REVIEW

**Comment by wafels** [comment]
Multiple Travis CIs barfed on setting up the build...

File: sunpy/map/compositemap.py [hunk 1]
```diff
@@ -431,14 +433,22 @@ def plot(self, axes=None, annotate=True,  # pylint: disable=W0613
...
+                if m.mask is None:
+                    ret.append(axes.imshow(m.data, **params))
+                else:
+                    ret.append(axes.imshow(np.ma.array(np.asarray(m.data), mask=m.mask), **params))
+            else:
+                # Check for the presence of masked map data
+                if m.mask is None:
+                    ret.append(axes.contour(m.data, m.levels, **params))
+                else:
+                    ret.append(axes.contour(np.ma.array(np.asarray(m.data), mask=m.mask), **params))
```
Comment (dpshelio) [inline sunpy/map/compositemap.py]: If `levels` exist, but there's also a mask... then the levels are not being used for the contour?

Comment (wafels) [reply sunpy/map/compositemap.py]: Ah, good catch!  Thanks!

**Comment by wafels** [comment]
Build fail is due to an online data acquisition test failing.

**Comment by wafels** [comment]
Also, this should be backported as it fixes a bug noted by a user.

**Comment by nabobalis** [comment]
FINE.

**Comment by nabobalis** [comment]
Closes #2490

**Comment by wafels** [comment]
😁

**Comment by nabobalis** [comment]
Now in 0.8.5. 

Remind me to move the changelog into master later in life.

**Pull request merged**